src/test/java/com/artipie/asto/key/KeyExcludeByIndexTest.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * The MIT License (MIT) Copyright (c) 2020-2021 artipie.com
+ * https://github.com/artipie/asto/LICENSE.txt
+ */
+package com.artipie.asto.key;
+
+import com.artipie.asto.Key;
+import org.hamcrest.MatcherAssert;
+import org.hamcrest.core.IsEqual;
+import org.junit.jupiter.api.Test;
+
+/**
+ * Test case for {@link KeyExcludeByIndex}.
+ *
+ * @since 1.9.1
+ */
+final class KeyExcludeByIndexTest {
+
+    @Test
+    void excludesPart() {
+        final Key key = new Key.From("1", "2", "1");
+        MatcherAssert.assertThat(
+            new KeyExcludeByIndex(key, 0).string(),
+            new IsEqual<>("2/1")
+        );
+    }
+
+    @Test
+    void excludesNonExistingPart() {
+        final Key key = new Key.From("1", "2");
+        MatcherAssert.assertThat(
+            new KeyExcludeByIndex(key, -1).string(),
+            new IsEqual<>("1/2")
+        );
+    }
```

Comment:
this behaviour is a bit unexpected for me. Probably at least document this one or throw an exception in such cases?

Reply:
@genryxy It said in the origin ticket this :
> If part to exclude was not found (or index is out of bounds), the class can return the origin key.
So, I think I will add it as documentation :)